types/api/category.d.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+type CategorySorting = '' | 'alpha' | 'manual' | 'recent'
+type CategoryType = 'channels' | 'direct_messages' | 'favorites' | 'custom'
+
+type CategoryOrder = string[]
+type CategoryChannelOrder = string[]
+
+type CategoryChannel = {
+    id?: string;
+    category_id: string;
+    channel_id: string;
+    sort_order: number;
+}
+
+type Category = {
+    id: string;
+    team_id: string;
+    display_name: string;
+    sort_order: number;
+    sorting: CategorySorting;
+    type: CategoryType;
+    muted: boolean;
+    collapsed: boolean;
+}
+
+type CategoryWithChannels = Category & {
+    channel_ids: CategoryChannelOrder;
```

Comment:
Ok, `CategoryChannelOrder` is an array of channel ids....perhaps we can delete `CategoryChannelOrder` and set it to string[] directly? Same for type CategoryOrder.

Reply:
I wanted to be specific here that it isn't just *any* array of strings.

Reply:
Why being specific about the array instead of the strings? If we had defined ChannelID as the type (even if it is retyping a string) having a ChannelID[] would be a lot more meaningful. Technically is the same thing, so I'm just wondering why being specific on that one and not on the ChannelID